lib/DXIL/DxilOperations.cpp

```diff
@@ -2737,8 +2737,8 @@ llvm::Attribute::AttrKind OP::GetMemAccessAttr(OpCode opCode) {
 }
 
 bool OP::IsOverloadLegal(OpCode opCode, Type *pType) {
-  DXASSERT(0 <= (unsigned)opCode && opCode < OpCode::NumOpCodes,
-           "otherwise caller passed OOB index");
+  if (opCode >= OpCode::NumOpCodes)
```

Comment:
```suggestion
  if (opCode == OpCode::NumOpCodes)
```

It is undefined behavior for a value of enumeration type to be out of range. (see my next suggestion for more on how to fix this case).

Reply:
Updated.